src/classes/ADDR_Addresses_TEST.cls

```diff
@@ -0,0 +1,562 @@
+        for (Contact c : cons) {
+            cityByContact.put(c.MailingCity, c); 
+        }
+        System.assertEquals(cityByAddress.get('City0').Id, cityByContact.get('City0').Current_Address__c);
+        System.assertEquals(cityByAddress.get('City0').MailingCity__c, cityByContact.get('City0').MailingCity);
+        
+        //Add filter on ADDR_Addresses_TDTM triggerHandler
+        for(TDTM_Global_API.TdtmToken token : tokens) {
+            if(token.className == 'ADDR_Addresses_TDTM') {
+                token.filterField = 'MailingCity__c';
+                token.filterValue = 'Brooklyn1';
+            }
+        }
+        
+        TDTM_Global_API.setTdtmConfig(tokens);
+        
+        Test.startTest();
+        delete addresses; 
+        Test.stopTest(); 
+        
+        List<Address__c> returnAddresses = [SELECT Id 
+                                            FROM Address__c 
+                                            WHERE Id IN :addresses]; 
+        System.assertEquals(0, returnAddresses.size()); 
+         
+        List<Contact> returnCons = [SELECT Id, Current_Address__c, MailingAddress, MailingCity 
+                                    FROM Contact 
+                                    WHERE Id IN :contacts];
+        Map<String, Contact> finalCityByContact = new Map<String, Contact>(); 
+        for (Contact con : returnCons) {
+            finalCityByContact.put(con.MailingCity, con); 
+        }
+        if (finalCityByContact.containsKey('NULL')) {
+            System.assertNOTEquals(NULL, finalCityByContact.get('NULL').MailingAddress);
+            System.assertEquals(NULL, finalCityByContact.get('NULL').Current_Address__c);
+        }
+ 
+    }
+    
+    /*********************************************************************************************************
+    * @description Test method that specifically verifies the following for afterDelete:
+    * (1) When an existing Address record is deleted with a Parent Account and it doesn't match the filter, Default_Address__c is set
+    * to true, the MailingAddress is cleared. 
+    * (2) When an existing Address record is deleted with a Parent Account and it does match the filter, Default_Address__c is set
+    * to true, the MailingAddress is not cleared. 
+    */  
+    @isTest
+    private static void afterDeleteWithParentAccount() {
+        UTIL_CustomSettingsFacade.getSettingsForTests(new Hierarchy_Settings__c(Account_Processor__c = UTIL_Describe_API.getHhAccRecTypeID(),
+                                                                               Contacts_Addresses_Enabled__c = true,
+                                                                               Household_Addresses_RecType__c = UTIL_Describe_API.getHhAccRecTypeID(),
+                                                                               Accounts_Addresses_Enabled__c = UTIL_Describe_API.getHhAccRecTypeID()));
```

Comment:
This doesn't look right here. Accounts_Addresses_Enabled__c is a boolean right? But UTIL_Describe_API.getHHAccRecTypeID() should return an Id? Bad copy pasta? Not sure how that worked.

Reply:
@lifewithryan, Accounts_Addresses_Enabled__c is misleading because it's actually not a checkbox but instead is a text area which contains the recordTypeId of the types of Account record types we want multi address to be enabled for.

Reply:
Hah, okay -- ignore me then. Could've sworn that was a different setting.